### DISCUSSION TITLE
How to identify if access token is a client token or a user token?

### DISCUSSION THREAD

**Discussion by sagarshah1983** [ory/hydra#2271] [Q&A] [answered]
Hello everyone,
Is there a good/ideal way to determine the token type (client token or a user token) in Ory Hydra? 
I am thinking of using and comparing client_id and sub field values and consider it a client token (if sub and client_id matches) else consider it as a user token. 
Is that a good approach?
Can I consider any other fields/claims from access token to derive to this?

Appreciate your suggestions!

**Comment by aeneasr** [comment, marked as answer]
The way you do it right now (comparison) is probably the only way to distinguish the token type!

**Reply by sagarshah1983** [reply]
Thank you @aeneasr for clarifications
